y2020/ex13/src/lib.rs

```diff
@@ -55,19 +55,15 @@ pub fn part2(input: &str) -> u64 {
     let mut candidate = max_bus.id - max_bus.offset - step;
 
     while !buses.is_empty() {
-        let mut satisfied_buses= vec![];
         candidate += step;
+
         // now we check if this creates a scale of timestamps
-        for (index, bus) in buses.iter().enumerate() {
-            if (candidate + bus.offset as u64) % bus.id as u64 == 0 {
-                step *= bus.id;  // here we should have done a LCM, but the bus ids are all prime, so...
-                satisfied_buses.push(index);
-            }
-        }
+        let (satisfied_buses, unsatified) : (Vec<Bus>, Vec<Bus>) = buses.into_iter()
+            .partition(|bus| (candidate + bus.offset as u64) % bus.id as u64 == 0);
 
-        for index in satisfied_buses.into_iter().rev() {
-            buses.remove(index);
-        }
+        step = satisfied_buses.iter().fold(step, |step, bus| step * bus.id);
```

Comment:
This PR benchmarks
time:   [8.9439 us 8.9656 us 8.9878 us]  

Solution in `main` benchmarks
time:   [3.4376 us 3.4563 us 3.4861 us]